packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -672,13 +688,14 @@ async function fluid({ file, args = {}, reporter, cache }) {
     })
   })
 
-  const base64Width = 20
+  const base64Width = options.base64Width || defaultBase64Width()
```

Comment:
can't you just use `options.base64Width`? It should be always set because of the healOptions function

Reply:
I am pretty sure I required it here but it's been a while since I wrote this so I'm not sure what situation required it.(**EDIT:** Answered)

Looking through the commits, this const was originally hardcoded with a value of 20. I then had `options.base64Width || 20` before moving the default value into a variable, and finally a function(to support a config option value in `gatsby.config.js`, otherwise falling back to the default value of 20).

`options.base64Width` is an optional param that can be set on the node/graphql function, should you want to use a different base64 width than the default value of 20. If you want to override the default globally via `gatsby.config.js`, `defaultBase64Width()` will return that value instead of 20.

At line 806, `base64Args` for the `fixed()` variant, `width` param is replaced with the `options.base64Width` if provided.

> It should be always set because of the healOptions function

Went over the code, the `fluid()` and `fixed()` functions initialize the `options` object at the start with no default params passed in. Only `generateBase64()` does this, where it originally passed in the hard coded default width of 20, my PR replaced that with `defaultBase64Width()` accordingly. That method is called by both fluid and fixed methods when they call `base64()` which first checks for a cached base64 data else it creates a new base64 image assigning the default width.

Fluid needs to know the width in advance for computing the height, this was the case prior to my PR as well.

Fixed derives it's height from the width during generation of base64 image afaik, I added a width param to it's base64 args only if a custom width was requested, otherwise it's provided a default width during base64 generation as mentioned above with `generateBase64`.

---

**TL;DR:** 
No, `options.base64Width` isn't handled by `healOptions()`, it's passed in from a param in the graphql query. 

For `fluid()` the width must be known to derive the `base64Height`, prior to this PR it was a fixed value, 20; now it can also be a configurable global value in `gatsby.config.js` too.

For `fixed()`, it's only interested in the graphql query param if available, otherwise it gets a default width from the 2nd `healOptions()` at a later stage in `generateBase64()`(assuming no cached version exists).

---

Should `options.base64Width` be changed to `args.base64Width` to avoid this confusion with `healOptions()`? Should any of what I've said be added as comments to document what is going on within the code more clearly?